### DISCUSSION TITLE
NPM_TOKEN and security

### DISCUSSION THREAD

**Discussion by robations** [pnpm/pnpm#4985] [General]
Hello! I've read in the past that there are potential security issues when installing packages with npm/yarn etc with postinstall scripts for dependencies reading sensitive environment variables (such as NPM_TOKEN), and it's necessary to use `--ignore-scripts` when installing, especially on CI servers.

Does anyone know if this is still relevant advice? Does it apply to pnpm?

**Comment by zkochan** [comment]
If you are concerned about this, you can use [pnpm.onlyBuiltDependencies](https://pnpm.io/package_json#pnpmonlybuiltdependencies) to list trusted packages that may run postinstall scripts.

In CI, I guess you can set NPM_TOKEN only for the step that does publish. But pnpm currently doesn't even read the token from NPM_TOKEN, it only reads the token from `.npmrc`
